vllm/core/block/prefix_caching_block.py

```diff
@@ -398,6 +420,13 @@ def is_full(self) -> bool:
     def num_empty_slots(self) -> int:
         return self._block.num_empty_slots
 
+    @property
+    def num_tokens_total(self) -> int:
+        if self._prev_block is None:
+            return len(self.token_ids)
+
+        return self._prev_block.num_tokens_total + len(self.token_ids)
```

Comment:
One throught here is to have a variable cache the compute result, so O(seq_len) is only for the first time, then later shall have O(1).
However this need an assumption that the chain won't break, so that previous num_token cached still hold valid, I'm not sure whether it is true in current V2 design?

Reply:
Got it, would apply similar design here.